post-merge update: block page (#7249)
ethereum/ethereum-org-website: src/content/developers/docs/blocks/index.md

![A diagram showing transaction in a block causing state changes](./tx-block.png)
_Diagram adapted from [Ethereum EVM illustrated](https://takenobu-hs.github.io/downloads/ethereum_evm_illustrated.pdf)_

> By spacing out commits, we give all network participants enough time to come to consensus: even though transaction requests occur dozens of times per second, blocks are only created and committed on Ethereum once every six seconds.

Reviewer: ```suggestion
By spacing out commits, we give all network participants enough time to come to consensus: even though transaction requests occur dozens of times per second, blocks are only created and committed on Ethereum once every twelve seconds.
```

## How blocks work {#how-blocks-work}

To preserve the transaction history, blocks are strictly ordered (every new block created contains a reference to its parent block), and transactions within blocks are strictly ordered as well. Except in rare cases, at any given time, all participants on the network are in agreement on the exact number and history of blocks, and are working to batch the current live transaction requests into the next block.

Once a block is put together by some validator on the network, it is propagated to the rest of the network; all nodes add this block to the end of their blockchain, and a new validator is selected to create the next block. The exact block-assembly process and commitment/consensus process is currently specified by Ethereum’s “proof-of-stake” protocol.

## Proof-of-stake protocol {#proof-of-work-protocol}

Proof-of-stake means the following:

> - Validating nodes have to stake at least 32 ETH into a deposit contract as collateral against bad behaviour. This helps protect the network because provably dishonet activity leads to some or all of that stake being destroyed.

Reviewer: ```suggestion
- Validating nodes have to stake 32 ETH into a deposit contract as collateral against bad behavior. This helps protect the network because provably dishonest activity leads to some or all of that stake being destroyed.
```

> - In every slot (spaced 6 seconds apart) a validator is randomly selected to be the block proposer. They bundle transactions together, execute them and determine a new 'state'. They wrap this information int a block and pass it around to other validators.

Reviewer: ```suggestion
- In every slot (spaced twelve seconds apart) a validator is randomly selected to be the block proposer. They bundle transactions together, execute them and determine a new 'state'. They wrap this information into a block and pass it around to other validators.
```

[More on mining](/developers/docs/consensus-mechanisms/pow/mining/)
[More on proof-of-stake](/developers/docs/consensus-mechanisms/pos)

## What's in a block? {#block-anatomy}

- `timestamp` – the time when the block was mined.
- `blockNumber` – the length of the blockchain in blocks.
- `baseFeePerGas` - the minimum fee per gas required for a transaction to be included in the block.
- `difficulty` – the effort required to mine the block.
- `mixHash` – a unique identifier for that block.
- `parentHash` – the unique identifier for the block that came before (this is how blocks are linked in a chain).
- `transactions` – the transactions included in the block.
- `stateRoot` – the entire state of the system: account balances, contract storage, contract code and account nonces are inside.

> - `nonce` – a hash that, when combined with the mixHash, proves that the block has gone through [proof-of-work](/developers/docs/consensus-mechanisms/pow/).

Reviewer: I think we should also show what's contained in the execution side of the block below the consensus side, of course taking out what is no longer relevant, perhaps making note of these changes. Would be helpful for developers.

Author: done!

> randao_reveal: a value used to select the next blockl proposer

Reviewer: ```suggestion
randao_reveal: a value used to select the next block proposer
```

> attestations: list of attestations in favour of the current block

Reviewer: ```suggestion
attestations: list of attestations in favor of the current block
```

sync_aggregate: subset of validators used to serve light clients
execution_payload: transactions passed from the execution client
```

Executing the transactions in the `execution_payload` updates the global state. All clients re-execute the transactions in the `execution_payload` to ensure the new state matches that in the new block `state_root` field. This is how clients can tell that a new block is valid and safe to add to their blockchain.

## Block time {#block-time}

> Block time refers to the time separating blocks. In Ethereum, time is divided up into 6 second units called 'slots'. In each slot a single validator is selected to propose a block. Assuming all validators are online and fully functional there will be a block in every slot, meanign the block time is 6s. However, occasionally validators might be offline when called to propose a block, meaning slots can sometimes go empty. This is different to proof-of-work based systems where block times are probabilistic and tuned by the mining difficulty.

Reviewer: ```suggestion
Block time refers to the time separating blocks. In Ethereum, time is divided up into twelve second units called 'slots'. In each slot a single validator is selected to propose a block. Assuming all validators are online and fully functional there will be a block in every slot, meaning the block time is 12s. However, occasionally validators might be offline when called to propose a block, meaning slots can sometimes go empty. This is different to proof-of-work based systems where block times are probabilistic and tuned by the mining difficulty.
```

## Block size {#block-size}

A final important note is that blocks themselves are bounded in size. Each block has a target size of 15 million gas but the size of blocks will increase or decrease in accordance with network demands, up until the block limit of 30 million gas (2x target block size). The total amount of gas expended by all transactions in the block must be less than the block gas limit. This is important because it ensures that blocks can’t be arbitrarily large. If blocks could be arbitrarily large, then less performant full nodes would gradually stop being able to keep up with the network due to space and speed requirements. The larger the block, the greater the computing power required to process them in time for the next slot. This is a centralizing force, which is resisted by capping block sizes.

## Further reading {#further-reading}

_Know of a community resource that helped you? Edit this page and add it!_

## Related topics {#related-topics}

- [Transactions](/developers/docs/transactions/)
- [Gas](/developers/docs/gas/)

> - [proof-of-stake](/developers/docs/consensus-mechanisms/pos)

Reviewer: ```suggestion
- [Proof-of-stake](/developers/docs/consensus-mechanisms/pos)
```
